mail/spring-cloud-starter-stream-source-mail/src/main/java/org/springframework/cloud/stream/app/mail/source/MailSourceProperties.java

```diff
@@ -0,0 +1,220 @@
+	 */
+	private String protocol;
+
+	/**
+	 * Username for email account. Usually is email id
+	 */
+	private String username;
+
+	/**
+	 * Password for email account.
+	 */
+	private String password;
+
+	/**
+	 * Host for email access
+	 */
+	private String host;
+
+	/**
+	 * Port to be used for email e.g. 993 for imaps
+	 */
+	private String port;
+
+	/**
+	 * Email folder to be scanned for email
+	 */
+	private String folder;
+
+	/**
+	 * Set to true to mark email as read.
+	 */
+	private boolean markAsRead = false;
+
+	/**
+	 * Set to true to delete email after download.
+	 */
+	private boolean delete = false;
+
+	/**
+	 * Set to true to use IdleImap Configuration.
+	 */
+	private boolean idleImap = false;
+
+	/**
+	 * JavaMail properties as a new line delimited string of name-value pairs, e.g. 'foo=bar\n baz=car'.
+	 */
+	private Properties javaMailProperties;
+
+	/**
+	 * @return the protocol
+	 */
+	public String getProtocol() {
+		return this.protocol;
+	}
+
+	/**
+	 * @param protocol the protocol to set
+	 */
+	public void setProtocol(String protocol) {
+		this.protocol = protocol;
+	}
+
+	/**
+	 * @return the username
+	 */
+	public String getUsername() {
+		return this.username;
+	}
+
+	/**
+	 * @param username the username to set
+	 */
+	public void setUsername(String username) {
+		this.username = username;
+	}
+
+	/**
+	 * @return the password
+	 */
+	public String getPassword() {
+		return this.password;
+	}
+
+	/**
+	 * @param password the password to set
+	 */
+	public void setPassword(String password) {
+		this.password = password;
+	}
+
+	/**
+	 * @return the host
+	 */
+	public String getHost() {
+		return this.host;
+	}
+
+	/**
+	 * @param host the host to set
+	 */
+	public void setHost(String host) {
+		this.host = host;
+	}
+
+	/**
+	 * @return the port
+	 */
+	public String getPort() {
+		return this.port;
+	}
+
+	/**
+	 * @param port the port to set
+	 */
+	public void setPort(String port) {
+		this.port = port;
+	}
+
+	/**
+	 * @return the folder
+	 */
+	public String getFolder() {
+		return this.folder;
+	}
+
+	/**
+	 * @param folder the folder to set
+	 */
+	public void setFolder(String folder) {
+		this.folder = folder;
+	}
+
+	/**
+	 * @return the markAsRead
+	 */
+	public boolean isMarkAsRead() {
+		return this.markAsRead;
+	}
+
+	/**
+	 * @param markAsRead the markAsRead to set
+	 */
+	public void setMarkAsRead(boolean markAsRead) {
+		this.markAsRead = markAsRead;
+	}
+
+	/**
+	 * @return the delete
+	 */
+	public boolean isDelete() {
+		return this.delete;
+	}
+
+	/**
+	 * @param delete the delete to set
+	 */
+	public void setDelete(boolean delete) {
+		this.delete = delete;
+	}
+
+	/**
+	 * @return the idleImap
+	 */
+	public boolean isIdleImap() {
+		return this.idleImap;
+	}
+
+	/**
+	 * @param idleImap the idleImap to set
+	 */
+	public void setIdleImap(boolean idleImap) {
+		this.idleImap = idleImap;
+	}
+
+	/**
+	 * @return the javaMailProperties
+	 */
+	public Properties getJavaMailProperties() {
```

Comment:
Right, it is optional and it is a `new Properties()` by default.
But it must be `not-null` anyway, since we can call `setJavaMailProperties(null);`
And it really can't be `null`.  See `Session` source code:
````java
// Constructor is not public
    private Session(Properties props, Authenticator authenticator) {
	this.props = props;
	this.authenticator = authenticator;

	if (Boolean.valueOf(props.getProperty("mail.debug")).booleanValue())
	    debug = true;
````